File: apps/spring-boot-api/src/main/java/org/bootstrapbugz/api/shared/error/exception/UnauthorizedException.java
package org.bootstrapbugz.api.shared.error.exception;

import lombok.Getter;
import org.springframework.http.HttpStatus;

import java.io.Serial;

@Getter
public class UnauthorizedException extends RuntimeException {
  @Serial private static final long serialVersionUID = -8525346226722308705L;
  private final HttpStatus status;

  public UnauthorizedException(String message) {
    super(message);
    this.status = HttpStatus.UNAUTHORIZED;
  }
}

File: backend/spring-boot-kotlin/src/main/kotlin/org/bootstrapbugz/api/shared/error/exception/UnauthorizedException.kt
package org.bootstrapbugz.api.shared.error.exception

import org.springframework.http.HttpStatus
import java.io.Serial

class UnauthorizedException(message: String?) : RuntimeException(message) {
    val status: HttpStatus = HttpStatus.UNAUTHORIZED

    companion object {
        @Serial
        private val serialVersionUID = -8525346226722308705L
    }
}
